front-end/src/components/Teacher/index.jsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+import { useLocation } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+// FontAwesome library
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faFacebookF,
+  faTwitter,
+  faInstagram,
+  faLinkedinIn,
+} from '@fortawesome/free-brands-svg-icons';
+import { faShareNodes } from '@fortawesome/free-solid-svg-icons';
+
+// Components
+import Tabs from 'components/common/Tabs';
+import Overview from 'components/Teacher/Overview';
+import Courses from 'components/Teacher/Courses';
+
+// Images
+import teacher from 'assets/images/teacher.jpg';
+import favourite from 'assets/images/favourite.png';
+import certificate from 'assets/images/certificate.png';
+import speechBubble from 'assets/images/speechBubble.png';
+import reward from 'assets/images/reward.png';
+
+// Styles
+import styles from './Teacher.module.scss';
+
+const Teacher = ({ fullName, activity, id, overview, courses }) => {
+  const { pathname } = useLocation();
+  const tabs = [
+    { title: 'Overview', link: `/app/teachers/${id}/overview` },
+    { title: 'Courses', link: `/app/teachers/${id}/courses` },
+  ];
+
+  return (
+    <div className={styles.wrapper}>
+      <div className={styles.test}>
+        <div className={styles.iconsWrap}>
+          <FontAwesomeIcon icon={faFacebookF} />
+          <FontAwesomeIcon icon={faTwitter} />
+          <FontAwesomeIcon icon={faInstagram} />
+          <FontAwesomeIcon icon={faLinkedinIn} />
+        </div>
+        <div className={styles.imageWrap}>
+          <img src={teacher} alt='teacher' />
+        </div>
+        <div className={styles.share}>
+          <FontAwesomeIcon icon={faShareNodes} />
+          <span>Report This Author</span>
+        </div>
+      </div>
+      <div className={styles.description}>
+        <h1>{fullName}</h1>
+        <span>{activity}</span>
+      </div>
+      <div className={styles.test1}>
+        <div className={styles.blockWrap}>
+          <img src={speechBubble} />
+          <span>533 Reviews</span>
```

Comment:
We need to add all mock teacher data in teacher collection and fetch  from back-end